src/libsync/discovery.cpp

```diff
@@ -1134,6 +1170,16 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
             item->_status = SyncFileItem::Status::NormalError;
         }
 
+        if (const auto folderQuota = folderQuotaAvailable(item);item->_size > folderQuota && folderQuota > -1) {
+            item->_instruction = CSYNC_INSTRUCTION_ERROR;
+            const auto folderName =
+            item->_errorString = tr("Upload of %1 exceeds the %2 of space left in %3 folder.", "folder sizes, folder name").arg(Utility::octetsToString(item->_size),
+                                                                                                Utility::octetsToString(folderQuota),
+                                                                                                _currentFolder._server.isEmpty() ? "root" : _currentFolder._server);
```

Comment:
for translations I would suggest to set a different error string entirely depending on this case, e.g.

```cpp
if (_currentFolder._server.isEmpty()) {
    item->_errorString = tr("Upload of %1 exceeds %2 of space left in the root folder.").arg(...);
} else {
    item->_errorString = tr("Upload of %1 exceeds %2 of space left in folder %3.").arg(...);
}
```

Reply:
in addition `root` is too technical
better use same wording from web UI: `Personal files`